datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -543,86 +543,17 @@ impl PhysicalExpr for BinaryExpr {
             )));
         }
 
+        // Attempt to use special kernels if one input is scalar and the other is an array
```

Comment:
There are no intended changes to this function's behavior, simply breaking it up into several smaller functions